client/battle/BattleActionsController.cpp

```diff
@@ -689,14 +689,14 @@ void BattleActionsController::actionRealize(PossiblePlayerBattleAction action, B
 				if (action.spell() == SpellID::SACRIFICE)
 				{
 					heroSpellToCast->aimToHex(targetHex);
-					possibleActions.push_back(PossiblePlayerBattleAction::SACRIFICE);
+					possibleActions.push_back({PossiblePlayerBattleAction::SACRIFICE, SpellID::SACRIFICE});
 					owner.stacksController->setSelectedStack(targetStack);
 					return;
 				}
 				if (action.spell() == SpellID::TELEPORT)
 				{
 					heroSpellToCast->aimToUnit(targetStack);
-					possibleActions.push_back(PossiblePlayerBattleAction::TELEPORT);
+					possibleActions.push_back({PossiblePlayerBattleAction::TELEPORT, SpellID::TELEPORT});

```

Comment:
I am looking for modernizing teleport effect (there is a TODO there, and I think we need teleport variant which will be affected by moats etc.), so, any hardcoded spellID is something than I look to avoid.

Reply:
Will remove one added by me (already available in `action` variable)

Reply:
Thanks)